packages/block-editor/src/components/block-list/block.js

```diff
@@ -416,12 +416,13 @@ function BlockListBlock( {
 			isFirstMultiSelected
 		);
 	const shouldShowMobileToolbar = ! isNavigationMode && shouldAppearSelected;
+	const isEditorEmpty = blockCount <= 1 && isEmptyDefaultBlock;
```

Comment:
I have been thinking about this and I think the issue might also happen if the first paragraph is empty but you have other blocks after it right?

I mean should the condition be something like this instead:

```
isFirstBlock && isEmptyDefaultBlock;
```

Reply:
Yes, and I tested all such cases and the solution works just fine. If there are multiple blocks, even if empty then it’s fine to have the appender.

Should I do with ‘isFirstBlock’?

Reply:
> If there are multiple blocks, even if empty then it’s fine to have the appender.

Why it's fine to have it for a multiple empty blocks but not when we have just one? In both cases, it seems that we can have the double inserter issue? I'll ping some designers to have thoughts here

@jasmussen @mapk 

I was also thinking that using `isFirstBlock` is better in terms of performance than using `blockCount` because the blocks won't rerender every time a block is added or removed.

Reply:
I tend to think we need to step back a little bit and think about the sibling inserter in a more holistic way. I do agree with #17416 in general, it is weird with the double inserter. However to me the solution isn't to add more edgecases. To that end I agree with Riad — simply reducing it to `isFirstBlock` should be good. An empty block is also a block, and one you might insert other blocks before. 

A rule of thumb should be that it should be obvious for users _why_ a thing appears, and subtextually, why it doesn't. It doesn't seem obvious why the sibling inserter would appear at the top when only the block isn't empty.

Reply:
I’m not on a laptop else would’ve made taken some screenshots to explain better but here’s my logic:

When you open the editor with the default block, it’s not a Paragraph block button more like an empty block with Inserter on the left.

But when you add another block below, even if empty, then the first block becomes a proper paragraph block without giving you an Inserter to change it to something else.

Hope that makes sense. 😀

Reply:
I get that an empty block is a block, but it doesn't _really_ act like a block. I can't move it, transform it, or see a block toolbar (for good reasons). So if one _other_ thing on an empty block works differently, that feels okay. In this case, I think if it's an empty block, it just doesn't allow the sibling inserter because the left inserter is already there next to it.

The other hand is that all those actions that an empty block is missing are specific to that block, while removing the sibling inserter is not. It's allowing an action to be made in the page... not in the empty block. So now I think I've come full circle. I can see leaving it in, especially if we improve the interaction with it.

Reply:
So what is the conclusion, if any? :)

Reply:
@mapk @jasmussen @youknowriad If an empty block is a block, then why can you "Add a block" on itself?

<img width="653" alt="Screen Shot 2019-10-01 at 4 44 56 PM" src="https://user-images.githubusercontent.com/3276087/66003112-d142e800-e46a-11e9-9d8d-7a5f7dc41b11.png">

The "empty block" is really but not really a paragraph block. But if we treat it more like a "block placeholder", then I would argue that it shouldn't have a prepending block sibling inserter, because no block has really been added. 

If, on the contrary, we agree that it is in fact a paragraph block, then I think it makes sense to have a prepending sibling inserter.

A little aside, but maybe related: why don't we show an appending sibling inserter for that first empty block?

Reply:
Also worth noting that when navigating with keyboard only, focus first goes to the appending sibling inserter and then to the editable area of the "empty block":

![2019-10-01 17-16-10 2019-10-01 17_17_37](https://user-images.githubusercontent.com/3276087/66004847-67790d00-e46f-11e9-97a3-91dd36adda67.gif)

Reply:
I'm willing to defer to whom has a strong opinion on this.

I think there's value in code simplicity, especially because many of these things are not yet 100% solidified, so we should not necessarily optimize towards these things staying the same. For example, we know that the Title _will_ become an actual block at some point. We might even make the placeholder an actual block. Does that change the equation?